contracts/Auction.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.5.0 < 0.9.0;
+
+contract Auction{
+    // who started auction, and will get paid
+    address payable public owner;
+    uint public startBlock;
+    uint public endBlock;
+
+    //off-chain info about item -- on IPFS
+    //keep pointer to IPFS hash
+    string public ipfsHash;
+
+    enum State {Started, Running, Ended, Canceled}
+    State public auctionState;
+
+    uint public highestBindingBid;
+    address payable public highestBidder;
+
+    // keeping addresses, amounts of bidders in a map
+    mapping (address => uint) public bids;
```

Comment:
what happens if one user bids on multiple items

Reply:
right.. my mental model for this exercise was that an auction was one entity with an owner for one item/batch per auction. obv.. need to discuss how this really should be modeled.

Reply:
https://www.tutorialspoint.com/solidity/solidity_events.htm